Sources/Amplitude/Types.swift

```diff
@@ -104,16 +104,15 @@ public enum PluginType: String, CaseIterable {
 
 public protocol Plugin: AnyObject {
     var type: PluginType { get }
-    var amplitude: Amplitude? { get set }
     func setup(amplitude: Amplitude)
     func execute(event: BaseEvent?) -> BaseEvent?
 }
 
 public protocol EventPlugin: Plugin {
-    func track(event: BaseEvent) -> BaseEvent?
```

Comment:
Minor, curious why we remove these (I can think up one case that the custom implementation of `track` might copy/update the `event` on the fly then return the updated one, this can be done in other ways too though.)

Reply:
@liuyang1520 
Result of destination plugins is ignored.
```
func processEvent(event: BaseEvent) {
    let beforeResult = self.applyPlugin(pluginType: PluginType.before, event: event)
    let enrichmentResult = self.applyPlugin(pluginType: PluginType.enrichment, event: beforeResult)
    _ = self.applyPlugin(pluginType: PluginType.destination, event: enrichmentResult)
}
```
https://github.com/amplitude/Amplitude-Swift/blob/main/Sources/Amplitude/Timeline.swift#L25

Are there EventPlugins that are not DestinationPlugins?
Maybe it makes sense to remove EventPlugin protocol and override DestinationPlugin methods `track`,`identify`,`groupIdentify`,`revenue`.

Reply:
AFAIK, currently all the EventPlugins are DestinationPlugins:
- Kotlin: https://github.com/search?q=repo%3Aamplitude%2FAmplitude-Kotlin%20EventPlugin&type=code
- Python: https://github.com/search?q=repo%3Aamplitude%2FAmplitude-Python%20EventPlugin&type=code
@justin-fiedler , @yuhao900914 , what do you think?

Reply:
Also changed `execute` signature from `execute(event: BaseEvent?) -> BaseEvent?` to `execute(event: BaseEvent) -> BaseEvent?`

Reply:
I think the change to `execute(event: BaseEvent) -> BaseEvent?` makes sense. Subsequent [plugins won't run on `nil` events](https://github.com/amplitude/Amplitude-Kotlin/blob/1d4477817e27444179691ba243e81a929be845a4/core/src/main/java/com/amplitude/core/platform/Mediator.kt#L21).

Since the Destination Plugins ignore the returned results, I agree it makes sense to either
1. Remove the return values for the EventPlugin methods
2. Process EventPlugins. Add separate handling of EventPlugins where we process the results. afaik right now adding a non-Destination EventPlugin will not be processed by the SDK.

In the interest of keeping the interface minimal, I vote for option 1 like Andrey implemented. In this case we should probably also [remove this logic in the Mediator](https://github.com/amplitude/Amplitude-Swift/blob/main/Sources/Amplitude/Mediator.swift#L28-L39) since it is [blocked up stream](https://github.com/amplitude/Amplitude-Swift/blob/main/Sources/Amplitude/Timeline.swift#L23-L25).

Reply:
I guess the EventPlugin originally is not just for Destination purpose, it was made to easier handling event and could be used for enrichment purpose, and destination plugin is more of a specific use of it. So I would suggest the use case for that before we remove the return values. I am not quite getting the process part not working though. Another concern for removing is consistency with other platforms.

Reply:
> it was made to easier handling event and could be used for enrichment purpose
Thanks, that makes sense. I think in this case we should keep the original return values.

> the event is not processed and passed now
My bad, missed EventPlugin is an interface, thought it could be add()ed directly.

Reply:
Thanks for the details Qinzhou. Agree on keeping `-> Bass Event?` for Event Plugin.

Reply:
Reverted `EventPlugin` logic